jib-core/src/main/java/com/google/cloud/tools/jib/cache/Caches.java

```diff
@@ -64,43 +54,59 @@ static void ensureOwnership(Path cacheDirectory)
         }
 
       } else {
-        // Creates the cache directory and ownership file.
-        Files.createDirectories(cacheDirectory);
-        Files.createFile(ownershipFile);
+        try {
+          // Creates the cache directory and ownership file.
+          Files.createDirectories(cacheDirectory);
+          Files.createFile(ownershipFile);
+
+        } catch (IOException ex) {
+          throw new CacheDirectoryCreationException(cacheDirectory, ex);
+        }
       }
     }
 
-    private final Path applicationCacheDirectory;
-    private Path baseCacheDirectory = DEFAULT_BASE_CACHE_DIRECTORY;
-
-    private Initializer(Path applicationCacheDirectory) {
-      this.applicationCacheDirectory = applicationCacheDirectory;
-    }
+    private final Path baseImageLayersCacheDirectory;
+    private final boolean shouldEnsureOwnershipOfBaseImageLayersCacheDirectory;
```

Comment:
Are these booleans only used for testing purposes, or do they have other practical uses?

Reply:
These are used in the `init` method below, which would only check for ownership by Jib if set to true.

Reply:
I was wondering if there was any reason to not check for Jib ownership, but I guess it's probably useful for library purposes

Reply:
When using a temporary directory, there's no need to check. Also, it would fail since the temporary directory has already been created.